source/containers.tex

```diff
@@ -14292,6 +14383,1709 @@
+    template<class K> iterator upper_bound(const K& x);
+    template<class K> const_iterator upper_bound(const K& x) const;
+
+    pair<iterator, iterator> equal_range(const key_type& x);
+    pair<const_iterator, const_iterator> equal_range(const key_type& x) const;
+    template<class K> pair<iterator, iterator> equal_range(const K& x);
+    template<class K> pair<const_iterator, const_iterator> equal_range(const K& x) const;
+
+    friend bool operator==(const flat_map& x, const flat_map& y);
+
+    friend @\exposid{synth-three-way-result}@<value_type>
+      operator<=>(const flat_map& x, const flat_map& y);
+
+    friend void swap(flat_map& x, flat_map& y) noexcept
+      { x.swap(y); }
+
+  private:
+    containers c;               // \expos
+    key_compare compare;        // \expos
+
+    struct key_equiv {  // \expos
+      key_equiv(key_compare c) : comp(c) { }
+      bool operator()(const_reference x, const_reference y) const {
+        return !comp(x.first, y.first) && !comp(y.first, x.first);
+      }
+      key_compare comp;
+    };
+  };
+
+  template<class KeyContainer, class MappedContainer>
+    flat_map(KeyContainer, MappedContainer)
+      -> flat_map<typename KeyContainer::value_type, typename MappedContainer::value_type,
+                  less<typename KeyContainer::value_type>, KeyContainer, MappedContainer>;
+
+  template<class KeyContainer, class MappedContainer, class Allocator>
+    flat_map(KeyContainer, MappedContainer, Allocator)
+      -> flat_map<typename KeyContainer::value_type, typename MappedContainer::value_type,
+                  less<typename KeyContainer::value_type>, KeyContainer, MappedContainer>;
+
+  template<class KeyContainer, class MappedContainer>
+    flat_map(sorted_unique_t, KeyContainer, MappedContainer)
+      -> flat_map<typename KeyContainer::value_type, typename MappedContainer::value_type,
+                  less<typename KeyContainer::value_type>, KeyContainer, MappedContainer>;
+
+  template<class KeyContainer, class MappedContainer, class Allocator>
+    flat_map(sorted_unique_t, KeyContainer, MappedContainer, Allocator)
+      -> flat_map<typename KeyContainer::value_type, typename MappedContainer::value_type,
+                  less<typename KeyContainer::value_type>, KeyContainer, MappedContainer>;
+
+  template<class InputIterator, class Compare = less<@\exposid{iter-key-type}@<InputIterator>>>
+    flat_map(InputIterator, InputIterator, Compare = Compare())
+      -> flat_map<@\exposid{iter-key-type}@<InputIterator>, @\exposid{iter-mapped-type}@<InputIterator>, Compare>;
+
+  template<class InputIterator, class Compare = less<@\exposid{iter-key-type}@<InputIterator>>>
+    flat_map(sorted_unique_t, InputIterator, InputIterator, Compare = Compare())
+      -> flat_map<@\exposid{iter-key-type}@<InputIterator>, @\exposid{iter-mapped-type}@<InputIterator>, Compare>;
+
+  template<ranges::input_range R, class Compare = less<@\exposid{range-key-type}@<R>>,
```

Comment:
Replace all `ranges::input_range` with `ranges::@\libconcept{input_range}@`.

Reply:
This is still missing.

Reply:
Done, thank you!